unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -4630,58 +4630,50 @@ public void PlaceHeldObject(string objectId, bool forceAction=false, bool placeS
             // #endif
         }
 
-        //used for all actions that need a sim object target
-        //instead of objectId, use screen coordinates to raycast toward potential targets
-        //will set the target object by reference if raycast is succesful
-        public bool ScreenToWorldTarget(float x, float y, ref SimObjPhysics target, bool requireWithinViewportRange)
-        {
-            //float x = action.x;
-            y = 1.0f - y; //reverse the y so that the origin (0, 0) can be passed in as the top left of the screen
-            //cast ray from screen coordinate into world space. If it hits an object
+        // used for all actions that need a sim object target
+        // instead of objectId, use screen coordinates to raycast toward potential targets
+        // will set the target object by reference if raycast is succesful
+        public bool ScreenToWorldTarget(float x, float y, ref SimObjPhysics target, bool requireWithinViewportRange) {
+            // reverse the y so that the origin (0, 0) can be passed in as the top left of the screen
+            y = 1.0f - y;
+
+            // cast ray from screen coordinate into world space. If it hits an object
             Ray ray = m_Camera.ViewportPointToRay(new Vector3(x, y, 0.0f));
             RaycastHit hit;
-            //if something was touched, actionFinished(true) always
-            if(Physics.Raycast(ray, out hit, Mathf.Infinity, 1 << 0 | 1 << 8 | 1 << 10, QueryTriggerInteraction.Ignore))
-            {
-                if(hit.transform.GetComponent<SimObjPhysics>())
-                {
-                    //wait! First check if the point hit is withing visibility bounds (camera viewport, max distance etc)
-                    //this should basically only happen if the handDistance value is too big
-                    if(requireWithinViewportRange && !CheckIfTargetPositionIsInViewportRange(hit.point))
-                    {
-                        errorMessage = "target sim object at screen coordinate: (" + x + ", " + y + ") is not within the viewport";
-                        return false;
+
+            // if something was touched, actionFinished(true) always
+            if (Physics.Raycast(ray, out hit, Mathf.Infinity, 1 << 0 | 1 << 8 | 1 << 10, QueryTriggerInteraction.Ignore)) {
+                if (hit.transform.GetComponent<SimObjPhysics>()) {
+                    // wait! First check if the point hit is withing visibility bounds (camera viewport, max distance etc)
+                    // this should basically only happen if the handDistance value is too big
+                    if (requireWithinViewportRange && !CheckIfTargetPositionIsInViewportRange(hit.point)) {
+                        throw new InvalidOperationException($"Target sim object at screen coordinate: ({x}, {y}) is not within the viewport");
```

Comment:
InvalidOperationException is C#'s built in way of handling Invalid State Exceptions.